pkg/controller/util/placement.go

```diff
@@ -86,3 +89,92 @@ func SetClusterNames(obj *unstructured.Unstructured, clusterNames []string) erro
 	}
 	return unstructured.SetNestedSlice(obj.Object, clusters, SpecField, PlacementField, ClustersField)
 }
+
+// ComputeNamespacedPlacement determines placement for namespaced
+// federated resources (e.g. FederatedConfigMap).
+//
+// If KubeFed is deployed cluster-wide, placement is the intersection
+// of the placement for the federated resource and the placement of
+// the federated namespace containing the resource.
+//
+// If KubeFed is limited to a single namespace, placement is
+// determined as the intersection of resource and namespace placement
+// if namespace placement exists.  If namespace placement does not
+// exist, resource placement will be used verbatim.  This is possible
+// because the single namespace by definition must exist on member
+// clusters, so namespace placement becomes a mechanism for limiting
+// rather than allowing propagation.
+func ComputeNamespacedPlacement(resource, namespace *unstructured.Unstructured, clusters []*fedv1b1.KubeFedCluster, limitedScope bool) (selectedClusters sets.String, err error) {
+	resourceClusters, err := ComputePlacement(resource, clusters)
+	if err != nil {
+		return nil, err
+	}
+
+	if namespace == nil {
+		if limitedScope {
+			// Use the resource placement verbatim if no federated
+			// namespace is present and KubeFed is targeting a
+			// single namespace.
+			return resourceClusters, nil
+		}
+		// Resource should not exist in any member clusters.
+		return sets.String{}, nil
+	}
+
+	namespaceClusters, err := ComputePlacement(namespace, clusters)
+	if err != nil {
+		return nil, err
+	}
+
+	// If both namespace and resource placement exist, the desired
+	// list of clusters is their intersection.
+	return resourceClusters.Intersection(namespaceClusters), nil
+}
+
+// ComputePlacement determines the selected clusters for a federated
+// resource.
+func ComputePlacement(resource *unstructured.Unstructured, clusters []*fedv1b1.KubeFedCluster) (selectedClusters sets.String, err error) {
+	selectedNames, err := selectedClusterNames(resource, clusters)
+	if err != nil {
+		return nil, err
+	}
+	clusterNames := getClusterNames(clusters)
+	return clusterNames.Intersection(selectedNames), nil
+}
+
+func selectedClusterNames(resource *unstructured.Unstructured, clusters []*fedv1b1.KubeFedCluster) (sets.String, error) {
+	placement, err := UnmarshalGenericPlacement(resource)
+	if err != nil {
+		return nil, err
+	}
+
+	selectedNames := sets.String{}
+	clusterNames := placement.ClusterNames()
+	// Only use selector if clusters are nil. An empty list of
+	// clusters implies no clusters are selected.
+	if clusterNames == nil {
```

Comment:
The problem with this approach is that RSP uses the clusternames itself to specify its decision, so the intersection of the cluster selector with the RSP decision will not work if both the cluster selector and the clusternames are specified on the target. 
Also, for the scenario where only the clusterselector is specified, the first reconcile result will be the intersection of the clusterselector and the RSP decision, but next time onwards the clusterselector will be ignored becaue RSP would have written clusternames on the target resource. So if the clusterselector is updated after that the update will be ignored.